### PR TITLE
Fix File Array passing on JES

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 1]
```diff
@@ -331,11 +333,22 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
...
-    def wdlFileToGcsPath(value: WdlValue, coerceTo: WdlType) =
-      if (coerceTo == WdlFileType && WdlFileType.isCoerceableFrom(value.wdlType))
-        jesOutputs find { _.name == value.valueString } map { j => WdlFile(j.gcs) } getOrElse value
-      else
-        value
+    def wdlValueToGcsPath(value: WdlValue): WdlValue = {
+      def toGcsPath(wdlFile: WdlFile) = {
+        jesOutputs find {
+          _.name == wdlFile.valueString
+        } map { j => WdlFile(j.gcs) } getOrElse value
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: onelineify, `map WdlFile`

File: src/main/scala/cromwell/binding/values/WdlMap.scala [hunk 2]
```diff
@@ -55,10 +55,17 @@ case class WdlMap(wdlType: WdlMapType, value: Map[WdlValue, WdlValue]) extends W
...
-      case (k: WdlPrimitiveType, v: WdlPrimitiveType) =>
+      case (wdlTypeKey: WdlPrimitiveType, wdlTypeValue: WdlPrimitiveType) =>
         Success(value.map({case (k, v) => s"${k.valueString}\t${v.valueString}"}).mkString("\n"))
       case _ =>
         Failure(new UnsupportedOperationException("Can only TSV serialize a Map[Primitive, Primitive]"))
     }
   }
+
+  def map(f: PartialFunction[((WdlValue, WdlValue)), (WdlValue, WdlValue)]): WdlMap = {
+    value.map{f} match {
```
Comment (mcovarr) [inline src/main/scala/cromwell/binding/values/WdlMap.scala]: mismatched dot and curlies

File: src/main/scala/cromwell/binding/values/WdlMap.scala [hunk 3]
```diff
@@ -55,10 +55,17 @@ case class WdlMap(wdlType: WdlMapType, value: Map[WdlValue, WdlValue]) extends W
...
+      case m:Map[WdlValue, WdlValue] if m.nonEmpty => WdlMap(WdlMapType(m.head._1.wdlType, m.head._2.wdlType), m)
```
Comment (mcovarr) [inline src/main/scala/cromwell/binding/values/WdlMap.scala]: space after colon

**Comment by mcovarr** [comment]
I assume this PR is for DSDEEPB-1765, but are the Map changes here related at all to DSDEEPB-1808?

**Comment by Horneth** [comment]
Ha yes, I didn't see that there was a ticket for Maps, but adding support for arrays it felt stupid not to add it for maps at the same time.

**Comment by Horneth** [comment]
So it actually does NOT fix the Map ticket after all, it adds missing pieces but it's not enough.

**Comment by Horneth** [comment]
https://github.com/broadinstitute/cromwell/pull/263 now replaces this PR

**Pull request closed without merging**